backend/src/Edna/Dashboard/DB/Query.hs

```diff
@@ -0,0 +1,154 @@
+import Edna.Util as U
+  (CompoundId, ExperimentId, MeasurementId, ProjectId, SqlId(..), SubExperimentId, TargetId,
+  fromSqlSerial, localToUTC)
+
+-- | Make given sub-experiment the primary one for its parent experiment.
+-- Previous primary sub-experiment is no longer primary.
+makePrimarySubExperiment :: SubExperimentId -> Edna ()
+makePrimarySubExperiment (SqlId subExpId) =
+  runUpdate' $ update (esPrimarySubExperiment ednaSchema)
+  (\pse -> pseSubExperimentId pse <-. val_ subExpId)
+  (\pse -> pseExperimentId pse ==. subquery_ (do
+    subExperiment <- all_ (esSubExperiment ednaSchema)
+    guard_ (seSubExperimentId subExperiment ==. val_ (SqlSerial subExpId))
+    return (seExperimentId subExperiment)
+    ))
+
+-- | Update name of a sub-experiment.
+setNameSubExperiment :: SubExperimentId -> Text -> Edna ()
+setNameSubExperiment (SqlId subExpId) name =
+  runUpdate' $ update (esSubExperiment ednaSchema)
+  (\se -> seName se <-. val_ name)
+  (\se -> seSubExperimentId se ==. val_ (SqlSerial subExpId))
+
+-- | Update @isSuspicious@ flag for a sub-experiment.
+setIsSuspiciousSubExperiment :: SubExperimentId -> Bool -> Edna ()
+setIsSuspiciousSubExperiment (SqlId subExpId) isSuspicious =
+  runUpdate' $ update (esSubExperiment ednaSchema)
+  (\se -> seIsSuspicious se <-. val_ isSuspicious)
+  (\se -> seSubExperimentId se ==. val_ (SqlSerial subExpId))
+
+-- | Delete a sub-experiment with given ID along with all @removed_measurement@
+-- entries for this sub-experiment.
+deleteSubExperiment :: SubExperimentId -> Edna ()
+deleteSubExperiment (SqlId subExpId) =
+  runDelete' $ delete (esSubExperiment ednaSchema) $
+  \se -> seSubExperimentId se ==. val_ (SqlSerial subExpId)
+
+-- | Check whether sub-experiment with given ID is the primary sub-experiment in
+-- any experiment.
+isPrimarySubExperiment :: SubExperimentId -> Edna Bool
+isPrimarySubExperiment (SqlId subExpId) =
+  fmap isJust . runSelectReturningOne' $ select $ do
+    primarySubExp <- all_ $ esPrimarySubExperiment ednaSchema
+    guard_ (pseSubExperimentId primarySubExp ==. val_ subExpId)
+    pure primarySubExp
+
+-- | Get data about all experiments using 3 optional filters: by project ID,
+-- compound ID and target ID.
+getExperiments :: Maybe ProjectId -> Maybe CompoundId -> Maybe TargetId ->
+  Edna [(ExperimentId, ExperimentResp)]
+getExperiments mProj mComp mTarget =
+  fmap (map convert . groupSubExps) $ runSelectReturningList' $ select $ do
+    experiment <- all_ $ esExperiment ednaSchema
+
+    experimentFile <- all_ $ esExperimentFile ednaSchema
+    guard_ (eExperimentFileId experiment ==. cast_ (efExperimentFileId experimentFile) int)
+
+    primarySubExp <- all_ $ esPrimarySubExperiment ednaSchema
+    guard_ (pseExperimentId primarySubExp ==. cast_ (eExperimentId experiment) int)
+
+    subExperiment <- leftJoin_ (all_ $ esSubExperiment ednaSchema) $
+      \subExp -> cast_ (seExperimentId subExp) int ==. eExperimentId experiment
```

Comment:
Do you mean inner join instead of left? Currently experiment always has at least one sub-experiment, so we can use inner join. My reasoning was that maybe one day we'll change something and it will be possible for an experiment to have no sub-experiments, in which case left join will be needed here.